R/visualize.R

```diff
@@ -0,0 +1,244 @@
+#'
+#' @param object ([PredictionClassif])\cr ([BenchmarkResult])\cr ([ResampleResult])\cr
+#' The binary class prediction object that will be evaluated. Only one data task is allowed for BenchmarkResult or ResampleResult
+#' * If provided a ([PredictionClassif]). Then the visualization will compare the fairness metrics among the binary level from protected field through bar plots.
+#' * If provided a ([ResampleResult]). Then the visualization will generate the boxplots for fairness metrics, and compare them among the binary level from protected field.
+#' * If provided a ([BenchmarkResult]). Then the visualization will generate the boxplots for fairness metrics, and compare them among both the binary level from protected field and the models implemented.
+#'
+#' @param ...
+#' The arguments to be passed to methods, such as:
+#'
+#' * `fairness_measure` ([Measure])\cr
+#' The fairness measures that will evaluated on object, could be single measure \link{msr} or measures \link{msrs}.  Default measure set to be \code{msr("fairness.acc")}
+#'
+#' * `task` ([TaskClassif])\cr
+#' The data task that contains the protected column, only required when the class of object is ([PredictionClassif])
+#'
+#' @examples
+#' library(mlr3fairness)
+#' library(mlr3learners)
+#' library(mlr3)
+#' library(ggplot2)
+#' library(data.table)
+#'
+#' # Setup the Fairness Measures and tasks
+#' task = tsk("adult_train")
+#' learner = lrn("classif.ranger", predict_type = "prob")
+#' learner$train(task)
+#' predictions = learner$predict(task)
+#' design = benchmark_grid(
+#'   tasks = tsk("adult_train"),
+#'   learners = lrns(c("classif.ranger", "classif.rpart"),
+#'                  predict_type = "prob", predict_sets = c("train", "test")),
+#'   resamplings = rsmps("cv", folds = 3)
+#' )
+#'
+#' bmr = benchmark(design)
+#' fairness_measure = msr("fairness.tpr")
+#' fairness_measures = msrs(c("fairness.tpr", "fairness.fnr"))
+#'
+#' fairness_compare(predictions, fairness_measure, task)
+#' fairness_compare(predictions, fairness_measures, task)
+#' fairness_compare(bmr, fairness_measure)
+#' fairness_compare(bmr, fairness_measures)
+#' @export
+fairness_compare <- function(object, ...){
+  UseMethod("fairness_compare")
+}
+
+#' @export
+fairness_compare.PredictionClassif <- function(object, fairness_measure = msr("fairness.acc"), task, ...){
+  measures = object$score(fairness_measure, task)
+  data <- melt(data.table(names = names(measures), data = measures), id.vars = "names")
+  ggplot(data, aes(x=names, y=value, fill = names)) +
+    geom_bar(stat = "identity",  width=0.4/length(unique(data$names))) +
+    xlab("metrics") +
+    ylab("values") +
+    theme(legend.position = "none") +
+    scale_fill_hue(c=100, l=60)
+}
+
+#' @export
+fairness_compare.BenchmarkResult <- function(object, fairness_measure = msr("fairness.acc"), ...){
+
+  if("list" %in% class(fairness_measure)) {
```

Comment:
This is not necessary, see https://github.com/mlr-org/mlr3/blob/main/R/as_measure.R#L49.

`as_measures(msr("classif.tpr"))` and `as_measures(msrs("classif.tpr"))` both work fine.

Reply:
I see what you mean, have updated the assert for all visualizations.